src/kernels/kernelRefreshIndicator.node.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { notebooks } from 'vscode';
+import { IExtensionSyncActivationService } from '../platform/activation/types';
+import { IPythonExtensionChecker } from '../platform/api/types';
+import { InteractiveWindowView, JupyterNotebookView } from '../platform/common/constants';
+import { disposeAllDisposables } from '../platform/common/helpers';
+import { IDisposable, IDisposableRegistry } from '../platform/common/types';
+import { IInterpreterService } from '../platform/interpreter/contracts';
+import { IKernelFinder } from './types';
+
+/**
+ * Ensures we refresh the list of Python environments upon opening a Notebook.
+ */
+@injectable()
+export class KernelRefreshIndicator implements IExtensionSyncActivationService {
+    private refreshedOnceBefore?: boolean;
+    private readonly disposables: IDisposable[] = [];
+    constructor(
+        @inject(IDisposableRegistry) disposables: IDisposableRegistry,
+        @inject(IPythonExtensionChecker) private readonly extensionChecker: IPythonExtensionChecker,
+        @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
+        @inject(IKernelFinder) private readonly kernelFinder: IKernelFinder
+    ) {
+        disposables.push(this);
```

Comment:
Is this a normal pattern to pass in a disposable registry? What's the purpose?

Reply:
this class is instantiated when the extension starts and it registers itself in to the list of disposables.
thus when the extension gets deactiavated this will get dispose


these are singletons, hence caller cannot dispose them untill the extension gets deactivated

Reply:
Ah ok, we could perhaps move this all into activationManager.ts by checking `'dispose' in item`?

Reply:
yes that's the suggestion @rebornix had